### PR TITLE
update LR scheduler section

### PR DESCRIPTION
**Pull request by abidwael**

### CODE REVIEW

**Review by jppgks — APPROVED** [review]

**Review by arnavgarg1 — APPROVED** [review]

File: docs/configuration/trainer.md [hunk 1]
```diff
@@ -19,18 +19,19 @@ By default, the ECD trainer is used.
...
         increase_batch_size_on_plateau_patience: 5
         increase_batch_size_on_plateau_rate: 2
         increase_batch_size_on_plateau_max: 512
-        decay: false
-        decay_steps: 10000
-        decay_rate: 0.96
-        staircase: false
+        learning_rate: 0.001
+        learning_rate_scheduler:
+            decay: false
```
Comment (tgaddair) [inline docs/configuration/trainer.md:28]: `decay: false` is now `decay: null` in the new schema.

I would also say this config is misleading, as all the decay params do nothing. We might want to think about making it better reflect a realistic config.

File: docs/configuration/trainer.md [hunk 2]
```diff
@@ -114,21 +115,22 @@ By default, the ECD trainer is used.
...
-    - `reduce_learning_rate_on_plateau` (default `0`): if theres a validation set, how many times to reduce the learning rate when a plateau of validation measure is reached.
-    - `reduce_learning_rate_on_plateau_patience` (default `5`): if theres a validation set, number of epochs of patience without an improvement on the validation measure before reducing the learning rate.
-    - `reduce_learning_rate_on_plateau_rate` (default `0.5`): if theres a validation set, the reduction rate of the learning rate.
+    - `learning_rate_scheduler` section:
+        - `reduce_on_plateau` (default `0`): if theres a validation set, how many times to reduce the learning rate when a plateau of validation measure is reached.
+        - `reduce_on_plateau_patience` (default `5`): if theres a validation set, number of epochs of patience without an improvement on the validation measure before reducing the learning rate.
+        - `reduce_on_plateau_rate` (default `0.5`): if theres a validation set, the reduction rate of the learning rate.
+        - `decay` (default `false`): if to use exponential decay of the learning rate or not.
```
Comment (tgaddair) [inline docs/configuration/trainer.md:122]: One of `null, exponential, linear`

File: docs/configuration/trainer.md [hunk 3]
```diff
@@ -114,21 +115,22 @@ By default, the ECD trainer is used.
...
+        - `decay_rate` (default `0.96`): the rate of the exponential learning rate decay.
+        - `decay_steps` (default `10000`): the number of steps of the exponential learning rate decay.
+        - `staircase` (default `false`): decays the learning rate at discrete intervals.
+        - `warmup_evaluations` (default `0`): Its the number or training epochs where learning rate warmup will be used. It is calculated as described in [Accurate, Large Minibatch SGD: Training ImageNet in 1 Hour](https://arxiv.org/abs/1706.02677). In the paper the authors suggest `6` epochs of warmup, that parameter is suggested for large datasets and big batches.
```
Comment (tgaddair) [inline docs/configuration/trainer.md:126]: Missing `warmup_fraction`.
